### PR TITLE
[optimize] Replace dubbo-demo-native

### PR DESCRIPTION
**Pull request by heliang666s**
## What is the purpose of the change?
#14878
1. Remove dubbo-demo-native
2. Modify dubbo-demo-springboot to support native functionality
3. Modify related github-action files

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [x] Make sure gitHub actions can pass. [Why the workflow is failing and how to fix it?](./CONTRIBUTING.md)

### CODE REVIEW

**Comment by AlbumenJ** [comment]
@CrazyHZM PTAL

**Review by FoghostCn — CHANGES_REQUESTED** [review]
native demo run failed, pls fix CI

File: dubbo-demo/dubbo-demo-spring-boot/dubbo-demo-spring-boot-provider/src/main/resources/application.yml [hunk 1]
```diff
@@ -21,23 +21,18 @@ spring:
...
     port: -1
+    serialization: fastjson2
   registry:
     id: zk-registry
     address: zookeeper://127.0.0.1:2181
   config-center:
     address: zookeeper://127.0.0.1:2181
   metadata-report:
     address: zookeeper://127.0.0.1:2181
-  metrics:
```
Comment (wcy666103) [inline dubbo-demo/dubbo-demo-spring-boot/dubbo-demo-spring-boot-provider/src/main/resources/application.yml:34]: Why remove this?

**Review by wcy666103 — COMMENTED** [review]
Why remove this?

**Pull request closed without merging**